hpedockerplugin/hpe_storage_api.py

```diff
@@ -463,50 +469,45 @@ def volumedriver_create(self, name, opts=None):
-                'promote' in contents['Opts']):
-            msg =(_('while reverting volume to snapshot status only valid '
-                    'option is promote=<vol_name>'))
-            LOG.error(msg)
-            return json.dumps({u"Err": six.text_type(msg)})
+            # Populating the values
+            if ('size' in contents['Opts']):
+                vol_size = int(contents['Opts']['size'])
 
-        # snapshotOf and cloneOf are mutually exclusive
-        if ('Opts' in contents and contents['Opts'] and
-                'snapshotOf' in contents['Opts'] and
-                'cloneOf' in contents['Opts']):
-            msg = (_('both snapshotOf and cloneOf cannot be specified at the '
-                     'same time'))
-            LOG.error(msg)
-            return json.dumps({u"Err": six.text_type(msg)})
+            if ('provisioning' in contents['Opts']):
+                vol_prov = str(contents['Opts']['provisioning'])
 
-        if ('Opts' in contents and contents['Opts'] and
-                'snapshotOf' in contents['Opts']):
-            return self.volumedriver_create_snapshot(name, opts)
-        elif ('Opts' in contents and contents['Opts'] and
-                'cloneOf' in contents['Opts']):
-            return self.volumedriver_clone_volume(name, opts)
+            if ('compression' in contents['Opts']):
+                compression_val = str(contents['Opts']['compression'])
 
-        vol_size = volume.DEFAULT_SIZE
-        if ('Opts' in contents and contents['Opts'] and
-                'size' in contents['Opts']):
-            vol_size = int(contents['Opts']['size'])
+            if ('flash-cache' in contents['Opts']):
+                vol_flash = str(contents['Opts']['flash-cache'])
 
-        vol_prov = volume.DEFAULT_PROV
-        if ('Opts' in contents and contents['Opts'] and
-                'provisioning' in contents['Opts']):
-            vol_prov = str(contents['Opts']['provisioning'])
+            if ('qos-name' in contents['Opts']):
+                vol_flash = str(contents['Opts']['qos-name'])
 
-        compression_val = volume.DEFAULT_COMPRESSION_VAL
-        if ('Opts' in contents and contents['Opts'] and
-                'compression' in contents['Opts']):
-            compression_val = str(contents['Opts']['compression'])
+            # check for valid promoteSnap option and return the result
+            if ('promote' in contents['Opts'] and len(contents['Opts']) == 1):
+                return self.revert_to_snapshot(name, opts)
+            elif ('promote' in contents['Opts']):
+                msg = (_('while reverting volume to snapshot status only '
+                         'valid option is promote=<vol_name>'))
+                LOG.error(msg)
+                return json.dumps({u"Err": six.text_type(msg)})
 
-        valid_compression_opts = ['true', 'false']
+            # mutually exclusive options check
+            mutually_exclusive_list = ['snapshotOf', 'cloneOf', 'qos-name']

```

Comment:
Please add promote to the mutually_exclusive_list as well

Reply:
Added 'promote' option.